src/request.rs

```diff
@@ -31,7 +31,14 @@ impl<'a> Request<'a> {
         let _ = self.stream.flush();
 
         let mut line = String::new();
-        try!(self.stream.read_line(&mut line));
+        match self.stream.read_line(&mut line) {
+            Ok(e) => {
+                if e == 0 {
+                    return Err(BeanstalkdError::ConnectionError)
+                }
+            },
+            Err(_) => return Err(BeanstalkdError::RequestError)
```

Comment:
So you might be changing the return type in the error case, and I'm wondering if that will have ripple effects. With this code, we're throwing away the old error code and always using `RequestError`. What would be closer to the old code would be `Err(e) => return Err(e)`

Reply:
And in the OK case, maybe also make it clearer what `e` represents and why it's an error for it to be 0

Reply:
I copied the code from the try macro that was present on this file - that returns BeanstalkdError::RequestError.
```
macro_rules! try {
    ($e:expr) => (match $e { Ok(e) => e, Err(_) => return Err(BeanstalkdError::RequestError) })
}
```
When I try to return `Err(e)` it complains of a mismatched type expected BeanstalkdError, found Error.

Reply:
Oh interesting! I assumed that this was the standard version of `try!` (which was a precursor to the `?` operator). Your approach makes sense, then.